datafusion/functions-nested/src/make_array.rs

```diff
@@ -198,7 +198,6 @@ pub(crate) fn make_array_inner(arrays: &[ArrayRef]) -> Result<ArrayRef> {
             let array = new_null_array(&DataType::Int64, length);
             Ok(Arc::new(array_into_list_array_nullable(array)))
         }
-        LargeList(..) => array_array::<i64>(arrays, data_type),
```

Comment:
I don't think this line was correct. According to the signature, the function output is always a List. However, this one sometimes returns a LargeList.

I added a test to illustrate this:
https://github.com/apache/datafusion/blob/91b468d0f337a73c65021c50ce889890cde9bccf/datafusion/sqllogictest/test_files/array.slt#L1160

Reply:
```
query ?
select make_array(make_array(1), arrow_cast(make_array(-1), 'LargeList(Int8)'));
----
[[1], [-1]]

query T
select arrow_typeof(make_array(make_array(1), arrow_cast(make_array(-1), 'LargeList(Int8)')));
----
List(Field { name: "item", data_type: LargeList(Field { name: "item", data_type: Int64, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }), nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} })

```

It seems correct 🤔 ?

Reply:
I see. You fix it :) Thanks